### DISCUSSION TITLE
AutoMap() not working / Vue 3

### DISCUSSION THREAD

**Discussion by bnar98** [nartc/mapper#581] [Q&A]
iam using the auto mapper like bellow , but the automap doesnt work , only ovjects in the custom mapping returned!

import type { RegionDto, UserDto } from "@/backend/auth";
import type { ServantDto, ServiceDto, TaskStatusEnum } from "@/backend/transport";
import { toUtc } from "@/composables/dateTimeFormat";
import { AutoMap, classes } from "@automapper/classes";
import { createMap, createMapper, forMember, mapFrom } from "@automapper/core";

const mapper = createMapper({ strategyInitializer: classes() });

export class TaskFilterData {
  id?: number;
  randomId?: number;
  regionId?: number;
  serviceId?: number;
  servantId?: number;
  clientUserId?: string;
  startAt?: string;
  endAt?: string;
  @AutoMap()
  status?: TaskStatusEnum;
  @AutoMap()
  page?: number;
}

export class TaskFilterDataDto {
  id?: number;
  randomId?: number;
  region?: RegionDto;
  service?: ServiceDto;
  servant?: ServantDto;
  user?: UserDto;
  startAt!: Date;
  endAt!: Date;
  @AutoMap()
  status?: TaskStatusEnum;
  page?: number;
}
createMap(
  mapper,
  TaskFilterDataDto,
  TaskFilterData,
  forMember(d => d.regionId, mapFrom(s => s.region?.id)),
  forMember(d => d.serviceId, mapFrom(s => s.service?.id)),
  forMember(d => d.servantId, mapFrom(s => s.servant?.id)),
  forMember(d => d.clientUserId, mapFrom(s => s.user?.uid)),
  forMember(d => d.startAt, mapFrom(s => toUtc(s.startAt))),
  forMember(d => d.endAt, mapFrom(s => toUtc(s.endAt))),

)
export function createMapTaskFilterDtoToTaskFilter(data: TaskFilterDataDto): TaskFilterData {
  return mapper.map(data, TaskFilterDataDto, TaskFilterData);
}

usnig with vue 3

**Comment by JamesPham2012** [comment]
for the automapper to automap Enums, I think you need to change the decorator like this
`
@AutoMap(() => String)
status?: TaskStatusEnum;
`

as for the `page` property, maybe you can try adding `@AutoMap()` decorator

**Reply by bnar98** [reply]
same thing for the page! i added @AutoMap() 
but still doesnt convert it , the only way i have to add it in the createMap function !!
